File: example/src/main/java/ru/virgil/example/order/BuyingOrderService.java
package ru.virgil.example.order;

import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Service;
import ru.virgil.example.system.SimpleJpaAccess;
import ru.virgil.example.user.UserDetails;

import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class BuyingOrderService implements SimpleJpaAccess<BuyingOrder, UUID> {

    private final BuyingOrderRepository repository;

    public List<BuyingOrder> getAll(UserDetails owner, int page, int size) {
        return repository.findAllByOwner(owner, PageRequest.of(page, size));
    }

    public BuyingOrder get(UserDetails owner, UUID uuid) {
        return repository.findByOwnerAndUuid(owner, uuid).orElseThrow();
    }

    public long countMy(UserDetails owner) {
        return repository.countAllByOwner(owner);
    }

    @Override
    public JpaRepository<BuyingOrder, UUID> getRepository() {
        return repository;
    }
}

File: example/src/main/java/ru/virgil/example/order/BuyingOrderService.kt
package ru.virgil.example.order

import org.springframework.beans.factory.ObjectProvider
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.data.domain.PageRequest
import org.springframework.stereotype.Service
import ru.virgil.example.user.UserDetails
import ru.virgil.example.user.UserDetailsService
import java.util.*

@Service
class BuyingOrderService(
    @Qualifier(UserDetailsService.current)
    private val ownerProvider: ObjectProvider<UserDetails>,
    private val buyingOrderRepository: BuyingOrderRepository,
) {

    private val owner by lazy { ownerProvider.getObject() }

    fun getAll(page: Int, size: Int): List<BuyingOrder> =
        buyingOrderRepository.findAllByOwner(owner, PageRequest.of(page, size))

    fun get(uuid: UUID): BuyingOrder = buyingOrderRepository.findByOwnerAndUuid(owner, uuid).orElseThrow()

    fun countMy(): Long = buyingOrderRepository.countAllByOwner(owner)
}
